packages/runtime/src/helpers/utils.ts

```diff
@@ -115,13 +127,33 @@ export const redirectsForNextRoute = ({
   status?: number
   force?: boolean
 }): NetlifyConfig['redirects'] =>
-  netlifyRoutesForNextRoute(route, buildId, i18n).map((redirect) => ({
+  netlifyRoutesForNextRoute(route, buildId, i18n).map(({ redirect }) => ({
     from: `${basePath}${redirect}`,
     to,
     status,
     force,
   }))
 
+export const redirectsForNext404Route = ({
```

Comment:
Could we add a unit test for this within `helpers/utils.spec`? Having one would have the benefit of shortening the feedback loop in case we make a change that breaks this specific functionality compared to the cypress integration test